src/src/fruit/hillel/Box.java

```diff
@@ -0,0 +1,53 @@
+package fruit.hillel;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Box<T extends Fruit> {
+    private List<T> fruits;
+
+    public Box() {
+        this.fruits = new ArrayList<>();
+    }
+
+    public void addFruit(T fruit) {
+        if (this.fruits.isEmpty() || fruit.getClass().equals(this.fruits.get(0).getClass())) {
+            this.fruits.add(fruit);
+        } else {
+            throw new IllegalArgumentException("You cannot add a fruit of a different type to the box.");
+        }
+    }
+
+    public void addFruits(List<T> fruits) {
+        for (T fruit : fruits) {
+            this.addFruit(fruit);
+        }
+    }
+
+    public float getWeight() {
+        float weight = 0.0f;
+        for (T fruit : this.fruits) {
+            if (fruit instanceof Apple) {
+                weight += 1.0f;
+            } else if (fruit instanceof Orange) {
+                weight += 1.5f;
+            }
```

Comment:
The box should know nothing about the particular type it is working with. Remove this and set the weight as a field in class `Fruit`, add a constructor to initialize it, and make orange and apple provide their weight to the superclass constructor. So here you just call something like `weight += fruit.getWeight()`